components/skus/renderer/skus_page_controller.cc

```diff
@@ -0,0 +1,353 @@
+    chrome_obj
+        ->Set(context, gin::StringToSymbol(isolate, "braveSkus"), skus_obj)
+        .Check();
+  } else {
+    skus_obj = skus_value->ToObject(context).ToLocalChecked();
+  }
+
+  // window.chrome.braveSkus.refresh_order
+  BindFunctionToObject(isolate, skus_obj, "refresh_order",
+                       base::BindRepeating(&SkusPageController::RefreshOrder,
+                                           base::Unretained(this), isolate));
+
+  // window.chrome.braveSkus.fetch_order_credentials
+  BindFunctionToObject(
+      isolate, skus_obj, "fetch_order_credentials",
+      base::BindRepeating(&SkusPageController::FetchOrderCredentials,
+                          base::Unretained(this), isolate));
+
+  // window.chrome.braveSkus.prepare_credentials_presentation
+  BindFunctionToObject(
+      isolate, skus_obj, "prepare_credentials_presentation",
+      base::BindRepeating(&SkusPageController::PrepareCredentialsPresentation,
+                          base::Unretained(this), isolate));
+
+  // window.chrome.braveSkus.credential_summary
+  BindFunctionToObject(
+      isolate, skus_obj, "credential_summary",
+      base::BindRepeating(&SkusPageController::CredentialSummary,
+                          base::Unretained(this), isolate));
+}
+
+template <typename Sig>
+void SkusPageController::BindFunctionToObject(
+    v8::Isolate* isolate,
+    v8::Local<v8::Object> javascript_object,
+    const std::string& name,
+    const base::RepeatingCallback<Sig>& callback) {
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  javascript_object
+      ->Set(context, gin::StringToSymbol(isolate, name),
+            gin::CreateFunctionTemplate(isolate, callback)
+                ->GetFunction(context)
+                .ToLocalChecked())
+      .Check();
+}
+
+// window.chrome.braveSkus.refresh_order
+v8::Local<v8::Promise> SkusPageController::RefreshOrder(v8::Isolate* isolate,
+                                                        std::string order_id) {
+  if (!EnsureConnected())
+    return v8::Local<v8::Promise>();
+
+  v8::MaybeLocal<v8::Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (resolver.IsEmpty()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
+  auto context_old(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+
+  skus_service_->RefreshOrder(
+      order_id,
+      base::BindOnce(&SkusPageController::OnRefreshOrder,
+                     base::Unretained(this), std::move(promise_resolver),
+                     isolate, std::move(context_old)));
+
+  return resolver.ToLocalChecked()->GetPromise();
+}
+
+void SkusPageController::OnRefreshOrder(
+    v8::Global<v8::Promise::Resolver> promise_resolver,
+    v8::Isolate* isolate,
+    v8::Global<v8::Context> context_old,
+    const std::string& response) {
```

Comment:
you should pass DictionaryValue directly, not a string `mojo_base.mojom.DictionaryValue`